src/types/field/tableField.ts

```diff
@@ -2,25 +2,23 @@ import { Document } from 'mongoose'
 
 import { IFormSchema } from '../form'
 
-import { IField, IFieldSchema } from './baseField'
+import { CoreField, IField, IFieldSchema } from './baseField'
 import { BasicField } from './fieldTypes'
 
 export interface IColumn {
+  _id?: string
   title: string
   required: boolean
-  // Allow all BasicFields, but pre-validate hook will block non-dropdown/
+  // Pre-validate hook will block non-dropdown/
   // non-textfield types.
-  columnType: BasicField
+  columnType: BasicField.Dropdown | BasicField.ShortText
+  fieldType?: BasicField.Dropdown | BasicField.ShortText //Optional fieldType so that IColumn can be passed into validateField
```

Comment:
This looks very hacky - we know a column field never has the `fieldType` property.